stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -261,6 +261,32 @@ extension Unsafe${Mutable}RawBufferPointer: ${Mutable}Collection {
 
 extension Unsafe${Mutable}RawBufferPointer: RandomAccessCollection { }
 
+extension Unsafe${Mutable}RawBufferPointer : ContiguousCollection {
+  @inlinable
+  @inline(__always)
+  public func withUnsafeBytes<R>(
+    _ body: (UnsafeRawBufferPointer) throws -> R
+  ) rethrows -> R {
+% if mutable:
+    return try body(UnsafeRawBufferPointer(self))
+% else:
+    return try body(self)
+% end # mutable
+  }
+
+  @inlinable
+  @inline(__always)
+  public func withUnsafeBufferPointer<R>(
+    _ body: (UnsafeBufferPointer<UInt8>) throws -> R
+  ) rethrows -> R {
+% if mutable:
+    return try body(UnsafeBufferPointer<UInt8>(self.bindMemory(to: UInt8.self)))
```

Comment:
Yikes! This API is outright incompatible with Swift's type safety. If it was legal to do this, then we would have no reason for `UnsafeRawPointer`.

Reply:
What makes this incompatible with Swift's type safety? Clearly my mental model is incorrect here — in my head I see this as the equivalent of `void *self; body((const uint8_t *)self);`; where is this model failing me?